recipes/sqlite3mc/all/conanfile.py

```diff
@@ -0,0 +1,225 @@
+        "enable_session": False,
+        "shell_is_utf8": True,
+        "enable_fileio": True,
+        "enable_regexp": True,
+        "enable_series": True,
+        "enable_sha3": True,
+        "enable_explain_comments": True,
+        "enable_dbpage_vtab": True,
+        "enable_dbstat_vtab": True,
+        "enable_stmtvtab": True,
+        "enable_unknown_sql_function": True,
+        "use_miniz": False,
+        "enable_compress": False,
+        "enable_sqlar": False,
+        "enable_zipfile": False,
+        "use_sqleet_legacy": False,
+        "use_sqlcipher_legacy": False,
+        "secure_memory": False,
+        "use_random_fill_memory": False,
+        "omit_aes_hardware_support": False
+    }
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.24 <4]")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if self.options.require_zlib:
+            self.requires("zlib/[>=1.2.11 <2]")
+        if self.options.with_icu:
+            self.requires("icu/74.1")
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
+
+        tc = CMakeToolchain(self)
+        tc.variables["_SQLITE3MC_REQUIRE_ZLIB"] = self.options.require_zlib
+        tc.variables["SQLITE3MC_STATIC_RUNTIME_LINK"] = self.options.static_runtime_link
+        tc.variables["SQLITE3MC_STATIC"] = not self.options.shared
+        tc.variables["SQLITE3MC_BUILD_SHELL"] = self.options.build_shell
+        tc.variables["SQLITE3MC_WITH_ICU"] = self.options.with_icu
+
+        tc.variables["SQLITE_ENABLE_DEBUG"] = self.options.enable_debug
+        tc.variables["SQLITE_SOUNDEX"] = self.options.soundex
+        tc.variables["SQLITE_ENABLE_COLUMN_METADATA"] = self.options.enable_column_metadata
+        tc.variables["SQLITE_SECURE_DELETE"] = self.options.secure_delete
+        tc.variables["SQLITE_ENABLE_FTS3"] = self.options.enable_fts3
+        tc.variables["SQLITE_ENABLE_FTS3_PARENTHESIS"] = self.options.enable_fts3_paranthesis
+        tc.variables["SQLITE_ENABLE_FTS4"] = self.options.enable_fts4
+        tc.variables["SQLITE_ENABLE_FTS5"] = self.options.enable_fts5
+
+        tc.variables["SQLITE_ENABLE_CARRAY"] = self.options.enable_carray
+        tc.variables["SQLITE_ENABLE_CSV"] = self.options.enable_csv
+        tc.variables["SQLITE_ENABLE_EXTFUNC"] = self.options.enable_extfunc
+        tc.variables["SQLITE_ENABLE_GEOPOLY"] = self.options.enable_geopoly
+        tc.variables["SQLITE_ENABLE_JSON1"] = self.options.enable_json1
```

Comment:
> I tried to find an usage for this option, but I failed to find it. I see it in the CMakeLists, but I don't see it in the code, I mean, a required definition like `ifdef`:
> 
> https://github.com/search?q=repo%3Autelle%2FSQLite3MultipleCiphers%20SQLITE_ENABLE_JSON1&type=code
> 
> Could you please explain a bit more how does it work, same to other options?

Several SQLite extensions can be included optionally at compile time using `#ifdef` statements. Mostly this happens in the main C file [sqlite3mc.c](https://github.com/utelle/SQLite3MultipleCiphers/blob/main/src/sqlite3mc.c) (like for `SQLITE_ENABLE_CSV` and `SQLITE_ENABLE_EXTFUNC`). 

But there are also extension that are already included in the original (resp patched) SQLite source amalgamation C file [sqlite3patched.c](https://github.com/utelle/SQLite3MultipleCiphers/blob/main/src/sqlite3patched.c) (like for `SQLITE_ENABLE_GEOPOLY`).

The symbol `SQLITE_ENABLE_JSON1` is different. This symbol does indeed not exist any more. In prior SQLite versions it was available, but in recent versions the JSON extension is included per default, but it could be explicitly excluded by specifying the symbol `SQLITE_OMIT_JSON`.

For the original project [**SQLite3 Multiple Ciphers**](https://github.com/utelle/SQLite3MultipleCiphers) I decided to expose many common SQLite configuration options to give users a high degree of freedom to choose the options and extensions most appropriate for their project. Per default many extensions are enabled, but in some projects there exist memory restrictions, so that users want to be able to disable unneeded extensions.